### PR TITLE
Fixes minimal image selection for amd64 on OCI.

### PR DESCRIPTION
**Pull request by tlm**
Due to the way we order images from the OCI provider there was a possability that under certain circumstances we could select a minimal Ubuntu image for amd64.

Currently Juju does not support minimal images and only has code to stop this from happening when the architecture was arm64.

This commit expands the catching methods to consider all minimal images regardless of arch.

At the moment we have only noticed this on @hpidcock machine when bootstrapping to the phoenix region. For that purpose we did a dump of some common image formats that are minimal so we can add them to unit tests for regression purposes.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Run unit tests in the OCI provider `go test github.com/juju/juju/provider/oci/...`. This will assert the regression tests.

Deploy a controller to OCI and check that the process is successful. When broken we expect the bootstrap to not succeed because of missing tools in the minimal image.

`juju bootstrap --config compartment-id=ocid1.compartment.oc1..aaaaaaaanvu2racnlczevenu73dlcf3nokgsjpdkbdgp4xbrz3lb2y4giyjq oci-canonical c`

## Documentation changes

N/A

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2051610

**Jira card:** JUJU-5388

### CODE REVIEW

**Comment by tlm** [comment]
#16795 is waiting on this work.

File: provider/oci/images.go [hunk 1]
```diff
@@ -318,34 +319,20 @@ func parseUbuntuImage(img ociCore.Image) (corebase.Base, string, error) {
...
-			// so we ignore the "Minimal aarch64" images.
-			return corebase.Base{}, "", errors.NotSupportedf("ubuntu minimal aarch64 image %s", *img.DisplayName)
-		} else {
-			// Else we assume the architecture is AMD64 by default.
-			arch = corearch.AMD64
-		}
+	if strings.Contains(*img.DisplayName, "Minimal") ||
+		strings.Contains(postfix, "Minimal") {
+		return corebase.Base{}, "", fmt.Errorf(
+			"%w ubuntu minimal image %q", errors.NotSupported, *img.DisplayName,
```
Comment (hpidcock) [inline provider/oci/images.go:329]: "not supported ubuntu minimal image bla bla" does not read well, I'd prefer "ubuntu minimal image %q %w"

File: provider/oci/images_integration_test.go [hunk 2]
```diff
@@ -7,10 +7,13 @@ import (
 	"context"
 	"time"
 
+	jc "github.com/juju/testing/checkers"
 	ociCore "github.com/oracle/oci-go-sdk/v65/core"
 	"go.uber.org/mock/gomock"
 	gc "gopkg.in/check.v1"
```
Comment (hpidcock) [inline provider/oci/images_integration_test.go:14]: delete blank line

**Review by hpidcock — APPROVED** [review]
Two minor problems. LGTM.

**Pull request merged**